src/utils/TokenUtils.js

```diff
@@ -211,14 +210,10 @@ define(function (require, exports, module) {
      * @return {mode:{Object}, name:string}
      */
     function getModeAt(cm, pos) {
-        var outerMode = cm.getMode(),
-            modeData = CodeMirror.innerMode(outerMode, cm.getTokenAt(pos, true).state),
```

Comment:
Yeah, you're right, `precise` defaults to `false`.
Apparently, `precise` only affects [`findStartLine`](https://github.com/codemirror/CodeMirror/blob/3cf2e251a3037089a626f53facfe91bfe65d790b/lib/codemirror.js#L1558-L1562), but I don't know to what extent.